### ISSUE TITLE
Fire an error when a media that is ready is not playing after X seconds

### ISSUE THREAD

**Issue by gordielachance** [cookpete/react-player#1382]
About medias that cannot be played (because they have a wrong video ID, or are geoblocked, ...),

#### Current Behavior

It seems that some players **do** fire an error when they cannot play the media (eg. Soundcloud) while others **don't** (eg. Youtube, as already mentionned [here](https://github.com/cookpete/react-player/issues/1067)) : so ReactPlayer is not consistent about this; and that makes it difficult to use is some circumstances.

#### Expected Behavior

1. If ReactPlayer is `playing={true}`; 
2. a timeout should be set when the media is _ready_; 
3. and cleared when the media _starts_ (or the URL changes)
4. If the timeout ends, it would fire an error.

Would this be possible please ?
Thanks for the amazing work !

**Comment by gordielachance** [comment]
I would like to help but I only learned React using hooks.
Any advises to start this out ?

**Comment by gordielachance** [comment]
Maybe some recent contributors could help ? @imjihun  @romain-daversa @korywka @komachi @Yasamato ?

**Comment by gordielachance** [comment]
Hello @cookpete, is it possible to get some help / tips to make this happen ?  Thanks !

**Comment by cookpete** [comment]
@gordielachance This should start you off:

https://codesandbox.io/s/lucid-currying-cf80ee?file=/src/App.js

I'm not even sure this needs to be added to ReactPlayer considering how simple it is just to implement it in the same place you are using `ReactPlayer`.

**Comment by gordielachance** [comment]
Thanks, I will test that !

> I'm not even sure this needs to be added to ReactPlayer considering how simple it is just to implement it in the same place you are using ReactPlayer.

I think it should.  A player that does not play should fire an error.  Since it does not play :)

If you use ReactPlayer to play a single URLs, that is maybe not a problem.  
But if, for me, you use it for playlists, it makes things very complicated.

**Comment by cookpete** [comment]
>I think it should. A player that does not play should fire an error. Since it does not play :)

Keep in mind that `onError` is literally just a proxy for the error callback of whatever library is being used to play the `url`. If we added it as a native ReactPlayer error we open doors such as: 

- How do we distinguish between a native ReactPlayer error and an error from a library?
- Who decides what the timeout is? What if it’s too long/short?
- Should the error still fire if the video isn’t played because of browser autoplay restrictions? How can we know that?
- etc

>But if, for me, you use it for playlists, it makes things very complicated.

Maybe a good feature to add to `react-playlister` then? 😉

**Comment by gordielachance** [comment]
Hi @cookpete, thanks for your reply!
I understand this is an "editorial" decision to take, and that you might decide not to implement it.

I'll try to make some tests based on the code you linked, but last time I made an attempt, I faced problems due to [player events not firing in the same order](https://github.com/cookpete/react-player/issues/1392).

> Who decides what the timeout is? What if it’s too long/short?

Developer decides.  Maybe the feature would be disabled (by default); if a timeout delay is not set.

> Should the error still fire if the video isn’t played because of browser autoplay restrictions? How can we know that?

No: in that case, the media _is_ playable - it's just blocked "locally" by the browser.
Also, there is a library available [here](https://github.com/video-dev/can-autoplay) to detect browser autoplay restrictions.
